core/src/main/scala/scalaz/Reducer.scala

```diff
@@ -157,10 +157,18 @@ sealed abstract class ReducerInstances {
     unitConsReducer(_ :: INil(), _ :: _)
   }
 
+  def ReverseIListReducer[C]: Reducer[C, IList[C]] = {
```

Comment:
instead of the additional Reverse reducers, it would be better to use the `@@ Tags.Dual` reducer. Though its implementation need to be updated to properly override `cons` and `snoc`.

Reply:
We don't even need all these new reverse reducers for the purposes of this PR. We could just use the existing `ReverseListReducer` and then reverse the reversed List into the appropriate data type (`IList`, `Stream`, `EphemeralStream`) using a simple `foldLeft`.

@jbgi's comment still stands for the existing `ReverseListReducer`, but that's for another PR.